spec/util/subscribeToResult-spec.ts

```diff
@@ -0,0 +1,156 @@
+    const expected = [1, 2, 3];
+    const result = Rx.Observable.range(1, 3);
+
+    const subscriber = new OuterSubscriber(x => {
+      expect(expected.shift()).to.be.equal(x);
+    }, () => {
+      done(new Error('should not be called'));
+    }, () => {
+      expect(expected).to.be.empty;
+      done();
+    });
+
+    subscribeToResult(subscriber, result);
+  });
+
+  it('should emits error when observable emits error', (done: MochaDone) => {
+    const result = Rx.Observable.throw(new Error('error'));
+    const subscriber = new OuterSubscriber(x => {
+      done(new Error('should not be called'));
+    }, (err) => {
+      expect(err).to.be.an('error', 'error');
+      done();
+    }, () => {
+      done(new Error('should not be called'));
+    });
+
+    subscribeToResult(subscriber, result);
+  });
+
+  it('should subscribe to array', (done: MochaDone) => {
+    const result = [1, 2, 3];
+    const expected = [];
+
+    const subscriber = new OuterSubscriber(x => expected.push(x),
+    () => {
+      done(new Error('should not be called'));
+    }, () => {
+      expect(expected).to.be.deep.equal(result);
+      done();
+    });
+
+    subscribeToResult(subscriber, result);
+  });
+
+  it('should subscribe to promise', (done: MochaDone) => {
+    const result = Promise.resolve(42);
+
+    const subscriber = new OuterSubscriber(x => {
+      expect(x).to.be.equal(42);
+    }, () => {
+      done(new Error('should not be called'));
+    }, done);
+
+    subscribeToResult(subscriber, result);
+  });
+
+  it('should emits error when promise rejects', (done: MochaDone) => {
+    const result = Promise.reject(42);
+
+    const subscriber = new OuterSubscriber(x => {
+      done(new Error('should not be called'));
+    }, (x) => {
+      expect(x).to.be.equal(42);
+      done();
+    }, () => {
+      done(new Error('should not be called'));
+    });
+
+    subscribeToResult(subscriber, result);
+  });
+
+  it('should subscribe to iterable', (done: MochaDone) => {
+    const iteratorResults = [
+      { value: 42, done: false },
+      { done: true }
+    ];
+
+    const iterable = {
+      [$$iterator]: () => {
+      return {
+        next: () => {
+          return iteratorResults.shift();
+        }
+      };
+    }};
+
+    const subscriber = new OuterSubscriber(x => {
+      expect(x).to.be.equal(42);
+    }, () => {
+      done(new Error('should not be called'));
+    }, done);
+
+    subscribeToResult(subscriber, iterable);
+  });
+
+  it('should subscribe to object has observable symbol', (done: MochaDone) => {
+    const observableSymbolObject = { [$$symbolObservable]: () => Rx.Observable.of(42) };
+
+    const subscriber = new OuterSubscriber(x => {
+      expect(x).to.be.equal(42);
+    }, () => {
+      done(new Error('should not be called'));
+    }, done);
+
+    subscribeToResult(subscriber, observableSymbolObject);
+  });
+
+  it('should emit error if observable symbol is not valid', (done: MochaDone) => {
```

Comment:
`emit an error if value returned by Symbol.observable call is not a valid observable`